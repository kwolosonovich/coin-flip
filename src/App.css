.App {
  text-align: center;
}

.App-header {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color:  #282c34;
  font-size: 40px;
  margin-top: 50px;
  margin-bottom: 50px;
}

