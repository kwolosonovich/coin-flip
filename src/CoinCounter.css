img {
    position: relative;
    display: block;
    margin: 50px auto;
    height: 200px;
    width: auto;

}

.score {
    text-align: center;
    margin-top: 20px;
}