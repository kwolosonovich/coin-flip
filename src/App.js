import React, { useState } from "react";
import CoinCounter from "./CoinCounter"
import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
          Coin Flip
      </header>
      <CoinCounter />
    </div>
  );
}

export default App;
