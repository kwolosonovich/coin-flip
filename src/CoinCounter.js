import React, {useState} from "react";
import App from "./App";
import Random from "./Random"
import './CoinCounter.css' 

// image source
let srcJpg = "https://static.thenounproject.com/png/98997-200.png";

const CoinCounter = () => {
    // set counter state [heads, tails]
    const [counter, setCounter] = useState([0,0])

    // set coin image state 
    const [isHeads, setIsHeads] = useState()

    // use random number to set heads to true or false
    const FlipCoin = () => {
        if (Random()) {
          // set heads image and increase tails counter
          setIsHeads(true);
          counter[0]++;
          setCounter([...counter])
          srcJpg = "https://2.bp.blogspot.com/-leqw-ggDo2E/W3km7SLGpxI/AAAAAAAAAMY/kKtdRLsT5mkUK2DRc1bme852a6WGLGC7QCLcBGAs/s400/dime-heads.jpg";
                    console.log(srcJpg);

        } else {
            // set tails image and increase tails counter
            setIsHeads(false);
            counter[1]++;
            setCounter([...counter]);
            srcJpg = "https://1.bp.blogspot.com/-axMqjO2fmug/W3km7ydbocI/AAAAAAAAAMg/VFmL2acMhAYyIillBSmo_KPr4rFMrIkUQCLcBGAs/s400/dime-tails.jpg";
        }
    }

    // display coin image 
    return (
      <>
        <button onClick={FlipCoin}>Flip Coin</button>
        <img className="coin" src={srcJpg} alt="coin" />

        <div className="score">
          You've flipped <b>{counter[0] + counter[1]}</b> times, with{" "}
          <b>{counter[0]}</b>
          heads and <b>{counter[1]}</b> tails!
        </div>
      </>
    );
}

export default CoinCounter; 