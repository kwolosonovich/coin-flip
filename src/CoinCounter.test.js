import React from "react";
import {
  render,
  fireEvent
} from "@testing-library/react";
import App from "./App";
import CoinCounter from "./CoinCounter";

// smoke test
it("renders without crashing", () => {
  render(<CoinCounter />);
});

// snapshot test
it("matches snapshot", () => {
  const { asFragment } = render(<CoinCounter />);
  expect(asFragment()).toMatchSnapshot()
});

beforeEach(function () {
  jest.spyOn(Math, "random").mockReturnValueOnce(0.25);
});

afterEach(function () {
  Math.random.mockRestore();
});

it("does not show coin", () => {
  const { queryByAltText } = render(<CoinCounter />);

  var img = queryByAltText("heads");
  expect(img).not.toBeInTheDocument();

  img = queryByAltText("heads");
  expect(img).not.toBeInTheDocument();
});

describe("test tails flip", () => {
  test("test tails", function () {
    jest.spyOn(Math, "random").mockReturnValueOnce(0.25);
  });

  it("flip coin to heads", () => {
    const { queryByAltText, getByText } = render(<CoinCounter />);

    fireEvent.click(getByText("Flip Coin"));

    let img = queryByAltText("tails flipped");
    expect(img).toBeInTheDocument();
  });
// });

// describe("test heads flip", () => {
  test("test heads", function () {
    jest.spyOn(Math, "random").mockReturnValueOnce(0.75);
  });

  it("flip coin to heads", () => {
    const { queryByAltText, getByText } = render(<CoinCounter />);

    fireEvent.click(getByText("Flip Coin"));

    let img = queryByAltText("tails flipped");
    expect(img).toBeInTheDocument();
  });
});
