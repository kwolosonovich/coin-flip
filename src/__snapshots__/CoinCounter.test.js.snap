// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <button>
    Flip Coin
  </button>
  <img
    alt="start"
    class="coin"
    src="https://static.thenounproject.com/png/98997-200.png"
  />
  <div
    class="score"
  >
    You've flipped 
    <b>
      0
    </b>
     times, with 
    <b>
      0
    </b>
     heads and 
    <b>
      0
    </b>
     tails!
  </div>
</DocumentFragment>
`;
